### PR TITLE
Lint module and first detector.

### PR DESCRIPTION
**Pull request by dtvc87**
The main propose of this PR is to set up the project to start using lint in order to simplify its usage as well as to give hints to its users.

This PR also includes the first detector `OptionalArgumentRequiredDetector`. This detector prohibits initializing a viewmodel optional argument using `SavedState.required()`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -23,14 +37,16 @@ buildscript {
         mockito                    : '2.28.2',
         mockitoKotlin              : '2.2.0',
         mockk                      : '1.9.3',
+        androidGradlePlugin        : androidGradlePlugin,
+        lint                       : lintVersion(androidGradlePlugin),
```
Comment (yamasa) [inline build.gradle]: Please replace `lint` with `androidLint`.

File: static-analysis/src/main/java/com/example/static_analysis/LichIssueRegistry.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.example.static_analysis
```
Comment (yamasa) [inline static-analysis/src/main/java/com/example/static_analysis/LichIssueRegistry.kt]: Please change the package name to `com.linecorp.lich.static_analysis`.

File: static-analysis/src/test/java/com/example/static_analysis/OptionalArgumentRequiredDetectorTest.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            getContent(TEST_DIR, filename = "ViewModelOptionalArgumentTestShowsError.kt")
+
+        lint().files(
+            kotlin(argumentDependencies),
+            kotlin(testWithErrors)
+        ).allowCompilationErrors(false)
+            .run()
+            .expect(
+                """
+                src/com/linecorp/lich/viewmodel/ArgumentTest.kt:5: Error: Optional argument initialized as required. [LichArgumentInitializationInconsistency]
```
Comment (yamasa) [inline static-analysis/src/test/java/com/example/static_analysis/OptionalArgumentRequiredDetectorTest.kt]: `LichArgumentInitializationInconsistency` should be `ArgumentInitializationInconsistency`?

**Comment by yamasa** [comment]
I merged `develop` into `master`. So, I've also changed the target branch of this PR to `master`.
Please rebase the PR, thanks!

File: static-analysis/src/main/java/com/example/static_analysis/LichIssueRegistry.kt (hunk 2 above)
Comment (dtvc87) [reply static-analysis/src/main/java/com/example/static_analysis/LichIssueRegistry.kt]: Right, I missed this, thanks!

File: static-analysis/src/test/java/com/example/static_analysis/OptionalArgumentRequiredDetectorTest.kt (hunk 3 above)
Comment (dtvc87) [reply static-analysis/src/test/java/com/example/static_analysis/OptionalArgumentRequiredDetectorTest.kt]: Good catch! thanks

**Comment by yamasa** [comment]
Thanks for your updates!

**Pull request merged**